client/authority-discovery/src/worker/tests.rs

```diff
@@ -385,62 +382,35 @@ fn terminate_when_event_stream_terminates() {
 	});
 
 	let (_to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
+	let worker = Worker::new(
 		from_service,
 		test_api,
 		network.clone(),
 		vec![],
 		dht_event_rx.boxed(),
-		Role::Authority(key_store),
+		Role::Authority(key_store.into()),
 		None,
 	);
 
-	block_on(async {
-		assert_eq!(Poll::Pending, poll!(&mut worker));
-
+	let timer = async {
+		Delay::new(Duration::from_secs(1)).await;
 		// Simulate termination of the network through dropping the sender side of the dht event
 		// channel.
 		drop(dht_event_tx);
+	};
+
+	let discovery_future = async {
+		let result = worker.run().await;
 
 		assert_eq!(
-			Poll::Ready(()), poll!(&mut worker),
+			(), result,
 			"Expect the authority discovery module to terminate once the sending side of the dht \
 			event channel is terminated.",
 		);
-	});
-}
-
-#[test]
-fn continue_operating_when_service_channel_is_dropped() {
-	let (_dht_event_tx, dht_event_rx) = channel(0);
-	let network: Arc<TestNetwork> = Arc::new(Default::default());
-	let key_store = KeyStore::new();
-	let test_api = Arc::new(TestApi {
-		authorities: vec![],
-	});
-
-	let (to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
-		from_service,
-		test_api,
-		network.clone(),
-		vec![],
-		dht_event_rx.boxed(),
-		Role::Authority(key_store),
-		None,
-	);
+	};
 
 	block_on(async {
```

Comment:
```suggestion
	block_on(join!(timer, discovery_future))
```

That should work as well, right?